components/allocator/lib.rs

```diff
@@ -21,7 +21,7 @@ mod platform {
     ///
     /// Passing a non-heap allocated pointer to this function results in undefined behavior.
     pub unsafe extern "C" fn usable_size(ptr: *const c_void) -> usize {
```

Comment:
Does this function need to be marked as unsafe if there's an `unsafe` block inside?

Reply:
Given that the safety requirement (heap allocated pointer) can't be checked inside the function, I would say it needs to be unsafe.